packages/hash/frontend/src/components/layout/MainContentWrapper.tsx

```diff
@@ -1,13 +1,75 @@
+import { IconButton, Fade, Box, Tooltip, styled } from "@mui/material";
 import { FunctionComponent } from "react";
+import { SIDEBAR_WIDTH } from "../../theme/components/navigation/MuiDrawerThemeOptions";
 
-import styles from "../../pages/index.module.scss";
+import { SidebarToggleIcon } from "../icons";
+import { HEADER_HEIGHT } from "./PageHeader/PageHeader";
 import { PageSidebar } from "./PageSidebar/PageSidebar";
+import { useSidebarContext } from "./SidebarContext";
+
+const Main = styled("main", {
+  shouldForwardProp: (prop) => prop !== "sidebarOpen",
+})<{
+  sidebarOpen?: boolean;
+}>(({ theme, sidebarOpen }) => ({
+  height: `calc(100vh - ${HEADER_HEIGHT}px)`,
+  overflowY: "scroll",
+  flexGrow: 1,
+  padding: "60px 120px 0 120px",
+  transition: theme.transitions.create("margin", {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  }),
+  marginLeft: `-${SIDEBAR_WIDTH}px`,
+  ...(sidebarOpen && {
+    transition: theme.transitions.create("margin", {
+      easing: theme.transitions.easing.easeOut,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    marginLeft: 0,
+  }),
+}));
 
 export const MainContentWrapper: FunctionComponent = ({ children }) => {
+  const { openSidebar, sidebarOpen } = useSidebarContext();
+
   return (
-    <div className={styles.MainWrapper}>
+    <Box
+      sx={{
+        display: "flex",
+        position: "relative",
+      }}
+    >
```

Comment:
@teenoh @akash-joshi fun fact, the following (equivalent) syntax exists:
```TypeScript
<Box display="flex" position="relative">
```
Using these props can significantly reduce the number of JSX code lines, and is also more readable because there's no `sx={{`.

Reply:
@benwerner01 arguably I would say this is more unreadable. CSS is generally grouped together, either in a CSS declaration, a `style` prop, or an `sx` prop in Mui's case.

In my opinion, the reduced number of lines of code aren't worth the extra time required by devs to grok the code (although I appreciate in this case it's pretty straightforward because there aren't any other props on the component).

Reply:
@akash-joshi didn't we also use `classname="flex"` when we were using tailwind? Would you have had the same objections then?

I think we might have to agree to disagree on this - so happy for this not to be changed in that case, as it does result in equivalent functionality.

However I'd argue that because we  agreed to use MUI, we shouldn't shy away from using MUI-specific convenience features (such as these props for the `Box` component), just because they look slightly different than the default way of doing things. I strongly disagree that this would "grok" the code. It's a feature of a library we agreed to use - so it's not unreasonable to expect developers to be able to read/use it (just as we'd expect them to use the MUI-specific `sx` prop, instead of putting things into the default `style` prop). In this case it's also really straightforward what a `position="relative"` prop would do, so I don't see how this would introduce any readability issues.

And I do think that the size of react trees can quickly become a readability issue in MUI when we're always chucking things like `display: "flex"` into the `sx` prop every time we want to use it, where it has to take up a whole line in the react tree. I'm pretty sure that's what the thinking behind adding these `Box` props was, because it let's these common styles be defined together in one line. 

I guess my thoughts boil down to: we agreed to use MUI, let's try and use it to it's full potential.

Reply:
@akash-joshi I find myself strongly agreeing with Ben on this. We have consciously bought into MUI now, so let's utilize their style fully.

Reply:
I think the benefit of using the inline syntax disappears once we have more than 3 styles.

Prettier will format this

```
 <Box display="flex" position="absolute" justifyContent="center" alignItems="center">
        <div />
 </Box>
```

to this

```
<Box
        display="flex"
        position="absolute"
        justifyContent="center"
        alignItems="center"
      >
        <div />
</Box>
```
which isn't so much different from using the `sx` prop directly. 

Although I understand that placing lots of styles in the `sx` prop can make the component a bit harder reader to read. I think we should adopt using [styled()](https://mui.com/system/styled/) for scenarios where we have a lot of styles in the sx prop

```
const MyComponent = styled('div')(({ theme }) => ({
  display: "flex",
  position: "absolute",
  justifyContent: "center",
  ...
}))


<MyComponent>
   <div />
</MyComponent>
```

We might lose some cool things like the shorthand syntax provided by the sx prop, but we can also use the [experimental_sx prop ](https://mui.com/system/styled/#how-can-i-use-the-sx-syntax-with-the-styled-utility)

Reply:
@teenoh granted when you add more than 3 of the props they wrap to the next line, but even in that case you're saving yourself two lines of code (`sx={{` and `}}`). And my instinct is we would be using 3 or less of them in most cases (just like in the example which started this thread).

I agree we should make more use of styled components, especially to create more reusable styling. However I don't think that would apply for this example: when all we're trying to do is add `display: "flex"` and `position: "relative"` **we can go from using 6 lines of code to just using 1** - and I think that is something worth doing throughout the codebase.

Reply:
@teenoh to be clear I agree with the example you gave. I'm thinking of the situations where there isn't a need for the `sx` prop because the inline props can be used instead. When we require styling (such as the `:hover` effect in your example) it doesn't make a difference if `display="flex"` is an inline prop, or defined in `sx`. In the example that started this thread however, it makes a big difference (we would go from 6 lines of code to 1).

Reply:
@benwerner01, just to be clear, I agree we should use inline props for components that have less than 4 style rules like in the example above. But for other scenarios, I think we should stick to an approach (inline props or sx/styled) for consistency and I'm in favour of the (sx/styled) approach

Reply:
Cool, thanks I think we're on the same page now. I think an easy rule of thumb would be that when we're already using the `sx` for something we shouldn't use inline props (as it makes no difference).